src/redux/auth/authSlice.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable camelcase */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const API_URL = 'http://localhost:3000/users/tokens';
+
+export const signup = createAsyncThunk('auth/signup', async (userData) => {
+  const response = await fetch(`${API_URL}/sign_up`, {
+    method: 'POST',
+    body: JSON.stringify(userData),
+    headers: { 'Content-Type': 'application/json' },
+  });
+  const data = await response.json();
+  return data;
+});
+
+export const signin = createAsyncThunk('auth/signin', async (userData) => {
+  const response = await fetch(`${API_URL}/sign_in`, {
+    method: 'POST',
+    body: JSON.stringify(userData),
+    headers: { 'Content-Type': 'application/json' },
+  });
+  const data = await response.json();
+  return data;
+});
```

Comment:
[OPTIONAL] you didn't put your code in a try and catch it is better to put code in try and catch so if the API has any error it shows to you and you can manage it to show a message to the user that something happened